### PR TITLE
fix: allow multiple apps to write safely to etc/environment

### PR DESCRIPTION
**Pull request by marcoppenheimer**
Blocked by https://github.com/canonical/zookeeper-operator/pull/62

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -132,3 +132,39 @@ def parse_tls_file(raw_content: str) -> str:
...
+    return map_env(env=raw_env)
+
+
+def update_env(env: dict[str, str]) -> None:
+    """Updates /etc/environment file."""
+    logger.info(f"{env=}")
+    logger.info(f"{get_env()=}")
+    updated_env = {**get_env(), **env}
+    logger.info(f"{updated_env=}")
+    content = "\n".join([f"{key}={value}" for key, value in updated_env.items()])
```
Comment (welpaolo) [inline src/utils.py:166]: If `updated_env` has the same content of `env` do we really need to rewrite the file ?

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -29,6 +29,40 @@
 DUMMY_NAME = "app"
 
 
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy_same_machine(ops_test: OpsTest, kafka_charm):
```
Comment (welpaolo) [inline tests/integration/test_charm.py:33]: Really nice to have an integration test for this!

**Review by welpaolo — APPROVED** [review]
LGMT, just left the same minor comment of the zookeeper charm

File: tests/integration/test_charm.py (hunk 2 above)
Comment (deusebio) [reply tests/integration/test_charm.py:33]: **praise** I agree!

**Review by deusebio — APPROVED** [review]
LGTM!

**Pull request merged**